test/test_sparse.py

```diff
@@ -1882,8 +1883,10 @@ def setUp(self):
         super(TestCudaSparse, self).setUp()
         self.is_cuda = True
         self.device = 'cuda'
-        self.IndexTensor = torch.cuda.LongTensor
-        self.ValueTensor = torch.cuda.DoubleTensor
+        self.value_dtype = torch.float64
+        self.value_empty = lambda *args: torch.empty(*args, dtype=self.value_dtype, device=self.device)
+        self.index_tensor = lambda *args: torch.tensor(*args, dtype=torch.int64, device=self.device)
+        self.value_tensor = lambda *args: torch.tensor(*args, dtype=self.value_dtype, device=self.device)
```

Comment:
I don't think we need to repeat this here, actually, because we capture `self` not `self.device` so the change to `self.device` will make it updated.